server/game/Interfaces.ts

```diff
@@ -30,6 +34,19 @@ export type IActionAbilityProps<TSource extends Card = Card> = Exclude<IAbilityP
     phase?: PhaseName | 'any';
 };
 
+export interface IOngoingEffectProps {
+    targetLocationFilter?: Location | Location[];
+    canChangeZoneOnce?: boolean;
+    canChangeZoneNTimes?: number;
```

Comment:
Do these need to be two properties? Also, where exactly will they be used? I can sort of imagine an effect granting a when defeated needing to persist into the discard; is that the kind of thing it's for?

Reply:
Oh yeah these are legacy, I'll either add a comment or remove them